### ISSUE TITLE
[ERROR]: Mismatch during LVS

### ISSUE THREAD

**Issue by akhilesh911** [The-OpenROAD-Project/OpenLane#1768]
### Description

Hello everyone,
I am new to OpenLANE and trying to rebuild a 16 bit processor using caravel user project. The repo is given [here](https://github.com/7vik-g/16bit-pipelined-RISC-processor-gf180.git). I am using gf180 PDK for this purpose. The design uses two macros, interface and processor, and four SRAM cells of gf180 PDK. There are mismatch in SRAM power pins, vdd and vss. But not such error in case of other macros. 

Net: processor/vdd                         |(no matching net)                          
  processor/vdd = 1                        |                                           
                                                       |                                           
Net: processor/vss                         |(no matching net)                          
  processor/vss = 1                        |                                           
---------------------------------------------------------------------------------------

---------------------------------------------------------------------------------------
Net: vss                                                                     |(no matching net)                          
  gf180mcu_fd_ip_sram__sram256x8m8wm1/VSS  |                                           
                                                                                  |                                           
Net: vdd                                                                     |(no matching net)                          
  gf180mcu_fd_ip_sram__sram256x8m8wm1/VDD  |                                           
The log files are attached. 
[23-lvs.lef.log](https://github.com/The-OpenROAD-Project/OpenLane/files/11260264/23-lvs.lef.log)
[23-user_project_wrapper.lef.lvs.log](https://github.com/The-OpenROAD-Project/OpenLane/files/11260266/23-user_project_wrapper.lef.lvs.log)

### Expected Behavior

No erros. Flow complete.

### Environment report

```shell
Kernel: Linux v5.4.0-146-generic
Distribution: ubuntu 18.04
Python: v3.6.9 (OK)
Container Engine: docker v20.10.17 (OK)
OpenLane Git Version: a35b64aa200c91e9eb7dde56db787d6b4c0ea12a
pip: INSTALLED
python-venv: INSTALLED
---
```

### Reproduction material

No reproducible issues found.

### Relevant log output

```shell
[INFO]: Writing Verilog (log: ../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/logs/signoff/21-write_powered_verilog.log)...
[INFO]: Running LVS (log: ../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/logs/signoff/23-lvs.lef.log)...
[ERROR]: There are LVS errors in the design: See '../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/reports/signoff/23-user_project_wrapper.lvs.rpt' for a summary and '../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/logs/signoff/23-lvs.lef.log' for details.
[INFO]: Saving current set of views in '../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/results/final'...
[INFO]: Generating final set of reports...
[INFO]: Created manufacturability report at '../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/reports/manufacturability.rpt'.
[INFO]: Created metrics report at '../home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/reports/metrics.csv'.
[INFO]: Saving runtime environment...
[ERROR]: Flow failed.
```

**Comment by vijayank88** [comment]
@akhilesh911 
Attach this log `/home/signoff/16bit-pipelined-RISC-processor-gf180/openlane/user_project_wrapper/runs/23_04_18_14_44/logs/signoff/23-lvs.lef.log`

**Comment by akhilesh911** [comment]
I've already attached.

**Comment by vijayank88** [comment]
@akhilesh911 
For gf180 check config.tcl: https://github.com/efabless/caravel_user_project/blob/gfmpw-0d/openlane/user_project_wrapper/config.tcl

`FP_PDN_MACRO_HOOKS` is
```
### Macro PDN Connections
set ::env(FP_PDN_MACRO_HOOKS) "\
	mprj vdd vss vdd vss"
```

**Comment by vijayank88** [comment]
@d-m-bailey 
For GF180 power domains are `vdd vss` right?

**Comment by akhilesh911** [comment]
@d-m-bailey I tried your suggestion. The pdn log gives warning that there is no via insertion between metal3 and metal4 for vdd and vss pins. Can you please suggest the solution?
[7-pdn.log](https://github.com/The-OpenROAD-Project/OpenLane/files/11292232/7-pdn.log)

**Comment by d-m-bailey** [comment]
@akhilesh911 probably not the best advice someone could give, but first try looking at the coordinates shown in the layout. Is there a metal3 power trunk in the macro that should connect to a metal4 power trunk in the top level?

Do you have metal4 obstruction layers over the macro?

Are your macros large enough or placed in the correct position that they intersect with the top power trunks?

**Comment by akhilesh911** [comment]
@d-m-bailey The gf180 SRAM macro doesnot have metal4 obstruction layers, but the processor macro have for both metal3 and metal4. 
The coordinates for macros are
micro_processor 1000 1900 N
IO_interface 600 750 N
instr_memory_MSB 1000 100 N
instr_memory_LSB 1550 100 N
data_memory_MSB 100 100 N
data_memory_LSB 2450 100 N

**Comment by d-m-bailey** [comment]
@akhilesh911 Sorry for the confusion. The pdn log has coordinates where vias cannot be placed. Can you open the gds and check those coordinates? You want to verify that metal3 and metal4 can be placed there. Does the SRAM macro get connected?

**Comment by vijayank88** [comment]
@d-m-bailey 
For a note, GF180 has power pins as `VDD VSS`. Is that to be updated in `FP_PDN_MACRO_HOOKS`?

**Comment by d-m-bailey** [comment]
@vijayank88 You may be right. It might be case sensitive.

**Comment by akhilesh911** [comment]
@vijayank88 @d-m-bailey Sorry for late replying. The signals are mapped correctly in verilog file. so there is no error for case sensitive.

**Comment by vijayank88** [comment]
@akhilesh911 Did you added `GRT_OBS` for SRAM macros?

**Comment by akhilesh911** [comment]
> @akhilesh911 Did you added `GRT_OBS` for SRAM macros?

@vijayank88 no, i haven't added that.

**Comment by d-m-bailey** [comment]
@akhilesh911 There should be a def file (and maybe a lef file) that you can open in the openroad gui, magic or maybe klayout. You should be able to check the error coordinates there.

**Comment by akhilesh911** [comment]
@d-m-bailey For me, openroad read_def shows no technology file has been read, even i have read_lib file. so i am unable to view def. And klayout is throwing error 
![deferror](https://user-images.githubusercontent.com/121286671/234195338-c5668706-e385-4d05-8f10-1fb66ee98995.png)

**Comment by kareefardi** [comment]
@akhilesh911 @vijayank88 is correct. 

> @d-m-bailey For a note, GF180 sram macros has power pins as `VDD VSS`. Is that to be updated in `FP_PDN_MACRO_HOOKS`?

`FP_PDN_MACRO_HOOKS` is a mapping between the design's power net(s) and a macro pin. Sram macro power pins are called `VDD` and `VSS`

**Comment by kareefardi** [comment]
In your case, the pin name of the sram macros match the standard cells pin names so "macro hooks" will be automatically done for the sram whether what you define is correct or not. What is actually missing is:
```
add_pdn_connect \
    -grid macro \
    -layers "Metal3 Metal4"
```
You need to create a copy of OpenLane's `scripts/openroad/common.tcl` and point to it using `PDN_CFG` - which will be renamed to match the documentation -. Then add the above code block to your newly created file. Currently OpenLane assumes fixed power layers for macros

**Comment by akhilesh911** [comment]
@kareefardi yes I understood. Along with this i have reconfigured some parameters, and the flow is passed. Thank you so much.

**Comment by vyom-elan** [comment]
I am implementing a sample timer project and during hardening the user_project_wrapper module I am getting a LVS mismatch error, I am not understanding the issue since hardening of user_project_example was successful. I am attaching the log file any input will be appreciated. Also I am getting several warning related to the PDK. Could this be a source for an error? 
![Screenshot from 2024-01-17 10-42-27](https://github.com/The-OpenROAD-Project/OpenLane/assets/97853828/2ae90eab-e7c7-4eab-958f-9f18c6708b5f)
Thank you

**Comment by vyom-elan** [comment]
LVS Log file of wrapper module - 
https://github.com/vyom-elan/caravel_uer_project/blob/main/user_project_wrapper.lef.lvs.log

**Comment by d-m-bailey** [comment]
@vyom-elan It looks there are inputs tied high or low on the top level of the design.

Since you are not synthesizing the top level, you'll get `conb` cells that aren't connected to power (no cell rows, no fill cells, no decap cells, no standard cell power rails).

There are a couple solutions.
1) Tie the inputs at the `user_proj_example` level.
2) Remove the `user_proj_example` level by flattening the design. This probably requires a little more work because you have to do a major overhaul of the `config.json` file. It's option 2 [here](https://caravel-harness.readthedocs.io/en/latest/caravel-with-openlane.html).

**Comment by vyom-elan** [comment]
@d-m-bailey  "It looks there are inputs tied high or low on the top level of the design." - I have interpreted it and checked it for any input set to logical high. Is that correct? Also I went with approach 2 and I am getting the following error during synthesis of user_project_wrapper. What could be the possible issue?
![Screenshot from 2024-01-23 12-47-21](https://github.com/The-OpenROAD-Project/OpenLane/assets/97853828/30f39113-0df5-497e-8852-d7079c7d26b7)
